llama_stack/apis/scoring_functions/scoring_functions.py

```diff
@@ -46,10 +46,11 @@ class CommonDef(BaseModel):
 
 
 @json_schema_type
-class DeterministicFunctionDef(CommonDef):
+class DeterministicFunctionDef(CommonFunctionDef):
```

Comment:
cc @raghotham about this name. this is the best we have for now to distinguish it from LLMJudgeFunctionDef. The obvious term "ScoringFunctionDef" is taken up by the Union type below so this needs to be a sub-type of that, so we need some kind of prefix. "Regular" is maybe another possibility, but it is also odd.

Reply:
Another thing is that we could technically define a braintrust's provider LLM As Judge scoring function using a `DeterministicFunctionDef`, so there's some ambiguity when choosing b/w "deterministic" v.s. "judge" types.

Reply:
@yanxi0830 yes totally. we need to kill Deterministic; let me think for a bit

Reply:
`ScoringFunctionDef` / `ScoringFunctionWithInferenceDef`?

- ScoringFunctionDef for general scoring functions & including braintrust LLMJudge
- ScoringFunctionWithInferenceDef for LLM judge scoring where the judge is a model needing LlamaStack inference

Reply:
Synced offline, updated to single `ScoringFunctionDef ` w/ optional `context` param.